lldb/source/Target/ThreadPlanStepOverBreakpoint.cpp

```diff
@@ -103,6 +103,12 @@ bool ThreadPlanStepOverBreakpoint::ShouldStop(Event *event_ptr) {
 
 bool ThreadPlanStepOverBreakpoint::StopOthers() { return true; }
 
+// The ThreadPlanSingleThreadTimeout can interrupt and resume all threads during
+// stepping, which may cause them to miss breakpoint. Therefore, we should
+// prevent the creation of ThreadPlanSingleThreadTimeout during a step-over
+// breakpoint.
```

Comment:
It might be a bit more clear to say something like:
```
// This thread plan does a single instruction step over a breakpoint instruction and needs
// to not resume other threads, so return false to stop the ThreadPlanSingleThreadTimeout
// from timing out and trying to resume all threads. If all threads gets resumed before we 
// disable, single step and re-enable the breakpoint, we can miss breakpoints on other 
// threads.
```